src/main/groovy/io/openliberty/tools/gradle/tasks/GenerateFeaturesTask.groovy

```diff
@@ -0,0 +1,348 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.openliberty.tools.gradle.tasks
+
+
+import io.openliberty.tools.common.plugins.config.ServerConfigXmlDocument
+import io.openliberty.tools.common.plugins.config.XmlDocument
+import io.openliberty.tools.common.plugins.util.BinaryScannerUtil
+import io.openliberty.tools.common.plugins.util.PluginExecutionException
+import io.openliberty.tools.common.plugins.util.ServerFeatureUtil
+import io.openliberty.tools.gradle.utils.ArtifactDownloadUtil
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.options.Option
+import org.xml.sax.SAXException
+import org.w3c.dom.Element;
+
+import javax.xml.parsers.ParserConfigurationException
+import javax.xml.transform.TransformerException
+import java.lang.reflect.InvocationTargetException
+
+class GenerateFeaturesTask extends AbstractFeatureTask {
+
+    private static final String GENERATED_FEATURES_FILE_NAME = "generated-features.xml";
+    protected static final String GENERATED_FEATURES_FILE_PATH = "configDropins/overrides/" + GENERATED_FEATURES_FILE_NAME;
+    protected static final String FEATURES_FILE_MESSAGE = "The Liberty Gradle Plugin has generated Liberty features necessary for your application in " + GENERATED_FEATURES_FILE_PATH;
+    protected static final String HEADER = "This file was generated by the Liberty Gradle Plugin and will be overwritten on subsequent runs of the generateFeatures task." + "\n It is recommended that you do not edit this file and that you commit this file to your version control.";
+    protected static final String GENERATED_FEATURES_COMMENT = "The following features were generated based on API usage detected in your application";
+    protected static final String NO_NEW_FEATURES_COMMENT = "No additional features generated";
+
+    private static final String BINARY_SCANNER_MAVEN_GROUP_ID = "com.ibm.websphere.appmod.tools";
+    private static final String BINARY_SCANNER_MAVEN_ARTIFACT_ID = "binary-app-scanner";
+    private static final String BINARY_SCANNER_MAVEN_TYPE = "jar";
+    private static final String BINARY_SCANNER_MAVEN_VERSION = "[21.0.0.4-SNAPSHOT,)";
+
+    private static final boolean DEFAULT_OPTIMIZE = true;
+
+    private File binaryScanner;
+
+    GenerateFeaturesTask() {
+        configure({
+            description 'Generate the features used by an application and add to the configuration of a Liberty server'
+            group 'Liberty'
+        })
+    }
+
+    private List<String> classFiles;
+
+    @Option(option = 'classFile', description = 'If set and optimize is false, will generate features for the list of classes passed.')
```

Comment:
Make sure to test with Gradle 7.x. We have had issues reported with task parameters not specifying `@Input` causing failures. See #643  and #667.

Reply:
Will add Gradle 7.x to test plan